driver-core/src/main/com/mongodb/internal/operation/FindOperation.java

```diff
@@ -797,6 +798,9 @@ private BsonDocument asDocument(final ConnectionDescription connectionDescriptio
         if (showRecordId) {
             document.put("$showDiskLoc", BsonBoolean.TRUE);
         }
+        if (allowDiskUse != null) {
```

Comment:
This shouldn't be here.  The option is only supported on find commands, and this method is for pre-find command support (3.0-).